src/environment.js

```diff
@@ -99,7 +97,14 @@ var Environment = Obj.extend({
         return this.filters[name];
     },
 
+    resolveTemplate: function(loader, parentName, filename) {
+        var isRelative = (loader.isRelative && parentName)? loader.isRelative(filename) : false;
+        return (isRelative && loader.resolve)? loader.resolve(parentName, filename) : filename;
```

Comment:
Looks good! One last nit: I think we can avoid checking is the methods exist and just call `isRelative` and `resolve` since they are on the base loader, and it this point we expect loaders to inherit from it.

So just `parentName ? loader.isRelative(filename) : false`, etc